api/envoy/config/rbac/v2alpha/rbac.proto

```diff
@@ -2,42 +2,49 @@ syntax = "proto3";
 
 import "validate/validate.proto";
 import "envoy/api/v2/core/address.proto";
-import "envoy/type/string_match.proto";
+import "envoy/api/v2/route/route.proto";
 
 package envoy.config.rbac.v2alpha;
 option go_package = "v2alpha";
 
-// Role Based Access Control (RBAC) provides service-level and method-level access control for a service.
-// The RBAC engine authorizes a request by evaluating the request context (expressed in the form of
-// :ref: `AttributeContext <envoy_api_msg_service.auth.v2alpha.AttributeContext>`) against the RBAC policies.
+// Role Based Access Control (RBAC) provides service-level and method-level access control for a
+// service. The RBAC engine authorizes a request by evaluating the request context (a combination of
+// the connection data and HTTP request headers) against the RBAC policies.
 //
-// RBAC policies are additive. The policies are examined in order. A request is allowed once a matching policy
-// is found (suppose the `action` is ALLOW).
+// RBAC policies are additive. The policies are examined in order. If the `action` is ALLOW, a
+// request is allowed once a policy is matched. If the `action` is DENY, a request is disallowed
+// once a policy is matched.
```

Comment:
To confirm my understanding, the top level `policy` match is always OR?
and if the `action` is ALLOW but the request it not matched (which means all the policies are not matched), then the request will be rejected, and vice-versa?
Thanks!

Reply:
that's correct:

A request is authorized IFF:
- Action=ALLOWED && a policy matches
- Action=DENY && no policy matches

A policy matches IFF:
- One (or more) permission matches the action
- One (or more) principal matches the downstream